src/ncp/ncp_base_mtd.cpp

```diff
@@ -682,8 +682,11 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_THREAD_ASSISTING_PORT
     otError         error      = OT_ERROR_NONE;
 
     // First, we need to remove all of the current assisting ports.
-    for (; numEntries != 0; ports++, numEntries--)
+    for (; numEntries != 0; numEntries--)
```

Comment:
Good find... while this fixes the issue, I think it'd be better that the implementation in `ncp_base` does not rely on the internal behavior/implementation and on how the ports are saved internally (i.e. how the content of the buffer `ports` variable is pointing to,  is changed by calling `otIp6RemoveUnsecurePort`)

Let's instead assume that when we get `otIp6GetUnsecurePorts`, the result become invalid if the list changes (i.e., an entry is added or removed). This is common model for iterators. We can add this in the documentation of `otIp6GetUnsecurePorts()`

Then we can use the following:
```
    // First, we need to remove all of the current assisting ports.
    ports = otIp6GetUnsecurePorts(mInstance, &numEntries)
    while (numEntries > 0) 
    { 
        SuccessOrExit(error = otIp6RemoveUnsecurePort(mInstance, *ports));   
        ports = otIp6GetUnsecurePorts(mInstance, &numEntries))
    }
```

Alternate option can that we add a new API to clear the entire unsecure port list, like `otIp6RemoveAllUnsecurePorts()` and then use it in `ncp_base_mtd.cpp`.

Reply:
I think it is better to add a new API `otIp6RemoveAllUnsecurePorts()`.  It will make the logic of the code clearer.